### ISSUE TITLE
Strange name of vaccine displayed.

### ISSUE THREAD

**Issue by vebaev** [covidpass-org/covidpass#52]
I just tried the update version, the romanised name works, but instead of Moderna says Spikevax which nobody knows what is it. Can you revert it back as before to say Moderna?

**Comment by PrincessRavy** [comment]
It seems that's the official name by Moderna now (https://www.ema.europa.eu/en/medicines/human/EPAR/spikevax-previously-covid-19-vaccine-moderna), maybe one could display both.

**Comment by marvinsxtr** [comment]
@vebaev The actual name is something along the lines of "Spikevax (previously COVID-19 Vaccine Moderna)", so I removed the part in the parentheses. "Moderna" would be the manufacturer, which is on the back of the pass.

**Comment by vebaev** [comment]
I think best is to write it Spikevax/Moderna or something?

**Comment by mackonsti** [comment]
As `Spikevax` is rather short (thankfully) the previously displayed (but very accurate!) `Spikevax (previously COVID-19 Vaccine Moderna)` was too long, one can argue that `Spikevax (Moderna)` can be a balanced choice that will still display OK and not too compressed...

**Comment by vebaev** [comment]
Spikevax (Moderna) Is good yes. They should know it but tell it to the grandma on my airport who checks the documents 😂

**Comment by marvinsxtr** [comment]
I would prefer to use the specification and not use custom names as these can become outdated pretty fast and are less accurate. A mix is also not preferable as Spikevax (Moderna Biotech Spain S.L.) would be too long to display.

**Comment by laurazard** [comment]
The specification is "Spikevax (previously COVID-19 Vaccine Moderna)". As you said so yourself, you already made a choice by removing the part in between parenthesis, which IMO is not the correct choice since 99% of people, including the people who might be checking these, will not have heard of Spikevax. As long as we're making arbitrary choices about how to display the information from the QR code, then any choice is as valid as another. I suggest something simple like "Spikevax (previously Moderna)", as to prevent confusion.

**Comment by marvinsxtr** [comment]
Removing the part in the parentheses was not really an arbitrary choice since the full name was simply too long to fit on the pass. Apart from that, I think custom names are feasible but require a certain amount of maintenance (Moderna is not the only vaccine with a strange new name/parentheses). Also, custom names would require us to hardcode some of the keys from the specification.

**Comment by laurazard** [comment]
Right, but my point is that it was a choice to omit the parenthesis part. I understand the full name was too long and I agree, but to solve that issue a choice was made to remove part of the value in the QR code, so either way we're already not sticking to the unaltered values. There aren't that many vaccine products out there so for now, if it were me I'd simply check for parenthesis with moderna inside and replace it with "Spiekvax (previously Moderna)" (I cloned the repo and did this, it looks fine). 

Regardless, I appreciate all the work you've put into this, I just think this issue might cause some complications for people trying to use this as the pass due to confusion that will inevitably happen.

**Comment by marvinsxtr** [comment]
I will reopen this and probably change the strange names to something short like Spikevax (Moderna) or Vaxzevria (AstraZeneca) in the future.

**Comment by vebaev** [comment]
I just already had days ago strange looks from the autoroutes at the airport, they wanted to see Moderna name and i had to open my original document :(

**Comment by marvinsxtr** [comment]
I don't really have experience using CovidPass in the wild yet so thanks for the feedback :)

**Comment by laurazard** [comment]
Thanks for taking our suggestions into consideration :)

**Comment by kaliGirl92277** [comment]
So what will go on passports? Using Relative Inheritance, Homeland should have implemented an AI Geocaching algorithm for monitoring vaccine administration points months ago.  Of course, we know that using arbitrary aliases is not the best solution. No doubt, that a Lack of efficacious AI code Inheritance and implementation by Homeland, combined with a fool pretending to exhibit executive leadership, who claimed precedence of self interest agendas using quid pro quo and plausible deniability (RTFM) over his priorities. Meanwhile, China is working on another bio weapon (COVID) variant. Maybe Spike will have some type of dramatic side effect. Why not just use mRNA-0. To indicate the number of variants the vaccine targets.

**Comment by kaliGirl92277** [comment]
Oh yes, COVID and all vector variants are definitely bio weapons inherited and engineered using Artificial Intelligence to determine propagate rates and optimal conditions for predetermined administration points to encourage the most rapid growth of the bio weapon. This() AI mechanism was initially intended as a gaming AI, created by China in 2019. RTFM! Observe ~ serendipity of specific dates pertaining to every single US Secretary of Defense appointment since 9/11. At least hackers have enough pride, humility, and failure to know what a sort algorithm is. Unfortunately, few exceptional programmers know how to read the output. Evidently.

**Comment by mackonsti** [comment]
> I will reopen this and probably change the strange names to something short like Spikevax (Moderna) or Vaxzevria (AstraZeneca) in the future.

We keep coming round and round in circles for this @marvinsxtr but indeed this is something that will satisfy both the new naming convention and keeping the old reference, too. Thanks again for your consideration.

**Comment by FrankyFFV** [comment]
From my point of view, should be "Spikevax (Moderna)".

**Comment by web2brain** [comment]
> From my point of view, should be:
> 
> 
> 
> Spikevax (Moderna)
> 
> Vaxzevria (AstraZeneca)
> 
> Comirnaty (BioNTech/Pfizer)

Janssen (Johnson & Johnson)

**Comment by luponata** [comment]
The naming convention is irrelevant,

The Pass will be scanned by a store/building employee using a **government application**, that is going to verify the data provided by the QR code, the person who scans has no authority on this, will only do what the application says, (you can enter the building or not)

The pass was not created with the purpose of competing with friends over who has the name of the coolest vaccine.

The only relevant information is the one on the QR code, as far as I'm concerned they could also write "Mickey Mouse", usually nobody would notice, neither could do anything about it.

It is certainly not our fault if other people do not know what Spikevax, Comirnaty, etc... are

I traveled across many europe countries without any problem

**Comment by laurazard** [comment]
I highly disagree. Yes, a lot of places will scan it and it will either show up as valid or not. However, a lot of places (restaurants, etc.) will just ask you to show the pass to look at it and see “if it looks right”. To these people, a name they don’t recognize might pose a problem. 

Moreover, as you mention your counties health ministry: in my country, the official certificate generated by the health ministry says Moderna, not Spikevax. However, when importing this certificate into the wallet with CovidPass, this name turns into Spikevax. This presents an incongruity between the name on “official” certificates and the data displayed on the generated pass, which is obviously not desirable.

**Comment by luponata** [comment]
> However, a lot of places (restaurants, etc.) will just ask you to show the pass to look at it and see “if it looks right”. To these people, a name they don’t recognize might pose a problem.

This is a different problem!, the Pass was not made to work this way!

They should have just left the QR code in sight **(with no other human readable writing)**, and this problem would have died in the bud.

Do you think that if one wanted to fake a certificate of any kind, he would not try to make it as similar to the original? wouldn't it be stupid to call the vaccine by a made-up name?

I can imagine it: 

**Vaccination certificate for Ligma-19**, (use "urban dict")
- Vaccines received: STFU dose 2 + GTFO dose 1

> in my country, the official certificate generated by the health ministry says Moderna, not Spikevax

Even in my country, during the process to receive the vaccine (on the day of the injection), citizens are sorted using these abbreviations: Moderna, Pfizer.

This is because the citizen in general does not enter into the merits, does not inquire or just does not care to know who made the vaccine, how many variants there are etc ...

It also allows governments not to generate confusion on the vaccination operation (since the pandemic has not already made things easier).

If they wanted to make it even simpler, they could unofficially call it: vaccine 1, vaccine 2, vaccine 3 ...

The health system knows very well which vaccine you have been given, this is the important thing.

When you tell your friends that you have hepatitis, the failure to specify the type of hepatitis does not put you in danger of life, if instead a doctor asks before giving you a drug then obviously yes

**LBNL**: The vaccine certificate must be **always scanned**, a person could just give you a fake or expired certificate and you might not know it by reading only the data on the sheet.

That's the "same" reason because the entire internet use SSL certificates, you can't rely on the FQDN alone to make sure someone isn't phishing your bank's website.
